cmd/infra/azure/create.go

```diff
@@ -868,3 +901,58 @@ func createLoadBalancer(ctx context.Context, subscriptionID string, resourceGrou
 	}
 	return nil
 }
+
+func assignContributorRole(subscriptionID, managedResourceGroupName, nsgResourceGroupName, vnetResourceGroupName, component, asigneeID string) error {
+	managedRG := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionID, managedResourceGroupName)
+	nsgRG := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionID, nsgResourceGroupName)
+	vnetRG := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionID, vnetResourceGroupName)
+
+	scopes := []string{managedRG}
+
+	cmdStr := fmt.Sprintf("az role assignment create --assignee-object-id %s --role \"Contributor\" --scope %s --assignee-principal-type \"ServicePrincipal\" ", asigneeID, managedRG)
+
+	switch component {
+	case cloudProvider:
+		scopes = append(scopes, nsgRG)
+	case ingress:
+		scopes = append(scopes, vnetRG)
+	}
+
+	for _, scope := range scopes {
+		cmdStr = fmt.Sprintf("az role assignment create --assignee-object-id %s --role \"Contributor\" --scope %s --assignee-principal-type \"ServicePrincipal\" ", asigneeID, scope)
+		_, err := execAzCommand(cmdStr)
+		if err != nil {
+			return fmt.Errorf("failed to assign contributor role to service principal, %s for scope %s : %w", asigneeID, scopes, err)
+		}
+		log.Log.Info("Successfully assigned contributor role to service principal", "ID", asigneeID, "scope", scope)
+	}
+
+	return nil
+}
+
+func findObjectId(appID string) (string, error) {
+	cmdStr := fmt.Sprintf("az ad sp show --id %s --query id | sed 's/\"//g'", appID)
+
+	output, err := execAzCommand(cmdStr)
+	if err != nil {
+		return "", fmt.Errorf("failed to find object ID for service principal, %s : %w", appID, err)
+
+	}
+	objectID := strings.ReplaceAll(string(output), "\n", "")
```

Comment:
That function returns the id based on the output from creating a sp and we can't refactor these commands since they're actively being used by our e2e tests